main/streams/plain_wrapper.c

```diff
@@ -1262,7 +1262,11 @@ static int php_plain_files_mkdir(php_stream_wrapper *wrapper, const char *dir, i
 		}
 		else {
 			/* find a top level directory we need to create */
-			while ( (p = strrchr(buf + offset, DEFAULT_SLASH)) || (offset != 1 && (p = strrchr(buf, DEFAULT_SLASH))) ) {
+			while ( (p = strrchr(buf + offset, DEFAULT_SLASH)) || (!(offset == 1
+#ifdef PHP_WIN32
+			 || (offset == 3 && IS_ABSOLUTE_WIN32_LOCAL_PATH(buf, offset))
+#endif
+			) && (p = strrchr(buf, DEFAULT_SLASH))) ) {
```

Comment:
I find it very hard to understand what is going on here. Can this be extracted into a helper function?

Reply:
@nikic Can you please propose the helper function? I just added an extra condition to the while loop to stop if root level of Windows path has been reached.

- the whole while condition checks if the path has more levels to strip and the path is not a root, but it is really specific for this part of the code, as it uses `oiffset` and also updates the `p` pointer.
- the `offset != 1` condition + the fix condition checks if the path is not a root, but only for normalized local file path (i.e. not UNC etc.)